extension/goplugin/environment.go

```diff
@@ -469,8 +471,8 @@ func handleEventForEnv(env IEnvironment, event string, requestContext map[string
 		for _, priority := range sortHandlers(prioritizedEventHandlers) {
 			for index, eventHandler := range prioritizedEventHandlers[priority] {
 				if err := eventHandler(requestContext, env); err != nil {
-					return fmt.Errorf("failed to dispatch event '%s' at priority '%d' with index '%d': %s",
-						event, priority, index, err)
+					env.Logger().Warningf("failed to handle event '%s' at priority '%d' with index '%d': %s", event, priority, index, err)
+					return goext.NewError(goext.ErrorBadRequest, err)
```

Comment:
err may already be a goext.Error. What happens in that case? I suggest modification of eventHandlers to always return goext.Error instead of error.

Reply:
fixed. Now, an extension can return either untyped error or typed goext.Error with some specific error code. Changing event handler declaration would be a massive backward compatibility break and would require many changes which IMO are not required.